platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -0,0 +1,290 @@
+of a brand-new network. The `config.txt` file, or at least its part that describes the address book components, will
+never be used in the life cycle of an existing network after that.
+
+### Startup procedure (pseudo-code)
+
+The pseudocode for the startup procedure will look as follows:
+
+```java
+if(a State
+exists on
+disk){
+
+loadStateFromDisk();
+}else{
+roster =
+
+loadFromConfigTxt();
+
+// Intall the roster as both Active Roster and Candidate Roster in the new state:
+createEmptyState(roster);
+// Note that this Active Roster is missing TSS key material.
+// So the node will only be able to sign using its RSA key, w/o any TSS.
+isGenesis =true;
+        }
+
+        if(
+the Statehas
+Candidate Roster){
+        // Check if this is a software upgrade, and if the TSS protocol
+        // has been launched already, then also check
+        // if Candidate Roster has enough TSS key material/signatures.
+        // Note that we switch to Candidate Roster during a software upgrade unconditionally
+        // until the TSS protocol is actually launched.
+        if(isSoftwareUpgrade /* && Candidate Roster is complete */){
+
+// This MUST be performed under `isSoftwareUpgrade` to ensure that
+// the entire network is being restarted, and so every node adopts the Candidate Roster,
+// and hence no ISSes happen.
+// Modify the state and put Candidate Roster into Active Roster, effectively clearing the Candidate Roster.
+makeCRtheAR();
+// May make a record of the previous Active Roster if necessary (e.g. for PCES replay)
+// Note that the previous Active Roster may lack TSS key material
+// if this is genesis, or if this is a TSS upgrade of an existing network.
+    }
+
+	/*
+	// This block will be uncommented once the TSS protocol is implemented
+	if (Candidate Roster is not complete) {
+		// Need to populate the Candidate Roster with TSS key material and signatures.
+		// Call the "TSS State Observer" to make it initiate the TSS protocol.
+		// See the Detecting a new Candidate Roster (aka the new “Platform API”) section below.
+		callTSSStateObserverToStartTSSProtocol();
+		// NOTE: processing of TSS messages will accumulate TSS key material
+		// in Candidate Roster and will emit a percentage metric indicating the readiness
+		// of the Candidate Roster for adoption. Once the metric is at 100%, this shows
+		// that the network can be restarted in order to adopt the Candidate Roster.
+	}
+	*/
+            }
+
+            if(
+the State
+has no
+Active Roster){
+
+// This should never happen
+throwFatalErrorAndShutdown();
+}
+
+// Check if the Active Roster is TSS-enabled
+        if(
+Active Roster
+is not
+TSS-enabled){
+  /*
+  // This block will be uncommented once the TSS protocol is implemented
+	if (isGenesis) {
+		rejectAnythingButTSSMessages = true;
+		// Note that a TSS upgrade of an existing network
+		// shouldn't disable non-TSS messages processing.	
```

Comment:
I don't follow how `rejectAnythingButTSSMessages` is set to true but there is a note about not disabling non-tss messages processing

Reply:
This flag is set under the `if (isGenesis)` condition. An existing network is past its genesis, and therefore the `isGenesis` will be false for existing networks. And therefore this `if()` branch won't execute, leaving the flag as `false`. The comment simply emphasizes this fact for clarity.

Reply:
Ok I understand, thanks for the explanation